app/models/proposal.rb

```diff
@@ -194,10 +194,12 @@ def partial_approve
   end
 
   def changed_fields comment_text
-    self.comments.create(
-      comment_text: comment_text, 
-      update_comment: true, 
-      user_id: self.requester_id
-    )
+    if !comment_text.blank?
```

Comment:
FYI that Rails also provides [`#present?`](http://api.rubyonrails.org/classes/Object.html#method-i-present-3F).